estate/models/estate_property.py

```diff
@@ -0,0 +1,123 @@
+            ("east", "East"),
+            ("west", "West"),
+            ("South", "South"),
+        ],
+        string="Garden Orientation",
+    )
+    state = fields.Selection(
+        [
+            ("new", "New"),
+            ("offer_received", "Offer Received"),
+            ("offer_accepted", "Offer Accepted"),
+            ("sold", "Sold"),
+            ("cancelled", "Cancelled"),
+        ],
+        string="Status",
+        required=True,
+        copy=False,
+        default="new",
+    )
+    active = fields.Boolean(default=True)
+    property_type_id = fields.Many2one("estate.property.type", "Property Type")
+    buyer_id = fields.Many2one("res.partner", "Buyer", copy=False)
+    salesperson_id = fields.Many2one("res.users", string="Salesperson")
+    tag_ids = fields.Many2many("estate.property.tag", string="Tags")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+    total_area = fields.Integer("Total Area (sqft)", compute="_compute_total_area")
+    best_price = fields.Float("Best Offer", compute="_compute_best_price")
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = (record.living_area or 0) + (record.garden_area or 0)
+
+    @api.depends("offer_ids.price")
+    def _compute_best_price(self):
+        for record in self:
+            record.best_price = max(record.offer_ids.mapped("price") or [0])
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+        for record in self:
+            if record.garden:
+                if record.garden_area == 10:
+                    record.garden_orientation = "north"
+                elif record.garden_area == 20:
+                    record.garden_orientation = "east"
+                elif record.garden_area == 30:
+                    record.garden_orientation = "west"
+                elif record.garden_area == 40:
+                    record.garden_orientation = "south"
+                else:
+                    record.garden_area = 0
+                    record.garden_orientation = False
+            else:
+                record.garden_area = 0
+                record.garden_orientation = False
+
+    def action_cancel(self):
+        for record in self:
+            if record.state == "sold":
+                raise UserError("A sold property cannot be cancelled")
+            record.state = "cancelled"
+
+    def action_sold(self):
+        for record in self:
+            if record.state == "cancelled":
+                raise UserError("A cancelled property cannot be set as sold")
+            record.state = "sold"
+
+    _check_expected_price = models.Constraint(
+        'CHECK(expected_price > 0)',
+        'The expected price must be strictly positive.'
+    )
+
+    _check_selling_price = models.Constraint(
+        'CHECK(selling_price > 0)',
+        'The selling price must be positive.'
+    )
```

Comment:
Constraints should be after the field declaration; you can take reference from the Odoo codebase.